lib/sensu/server/process.rb

```diff
@@ -572,6 +575,58 @@ def determine_stale_clients
         end
       end
 
+      # Determine stale checks, those that have not sent an event
+      # in a specified amount of time (thresholds). This method
+      # iterates through the result registry. If the time since
+      # the latest event is equal to or greater than a threshold,
+      # the check `:output` is set to a descriptive message, and
+      # `:status` is set to the appropriate non-zero value, and an
+      # alert is sent. If a check has been sending events, then
+      # no alert is published.
+      def determine_stale_checks
+        @logger.info("determining stale checks")
+        @redis.smembers("clients") do |clients|
+          clients.each do |client_name|
+            @redis.keys("result:" + client_name + ":*") do |results|
+              results.each do |result_name|
+                next if result_name.split(':')[2] == 'keepalive'
+                @redis.get(result_name) do |result_json|
+                  next if result_json.nil?
+                  result = MultiJson.load(result_json)
+                  check_defaults = {
+                    :thresholds => {
```

Comment:
But one of the *best* things about sensu is I can use configuration management to add and remove checks at will? If this is automatically added to every check, then won't it mean it will require manual intervention (or additional tooling) to auto-resolve all the checks that I "comment out" of my CM tool?


Additionally, this is a dramatic change in "before" behavior, where checks get a "natural" host dependency. If I have a host with 100 standalone checks, and it goes down, I'll get 100 alerts unless I *also* add an explicit host keepalive dependency. This should at least be noted. (and something I personally object to, I vote that this be opt-in)